network/client.go

```diff
@@ -18,30 +18,41 @@ const (
 )
 
 type Client struct {
-	runenv *runtime.RunEnv
-	client sync.Client
+	runenv     *runtime.RunEnv
+	syncClient sync.Client
 }
 
 // NewClient returns a new network client. Use this client to request network
 // changes, such as setting latencies, jitter, packet loss, connectedness, etc.
-func NewClient(client sync.Client, runenv *runtime.RunEnv) *Client {
+func NewClient(syncClient sync.Client, runenv *runtime.RunEnv) *Client {
 	return &Client{
-		runenv: runenv,
-		client: client,
+		runenv:     runenv,
+		syncClient: syncClient,
 	}
 }
 
 // WaitNetworkInitialized waits for the sidecar to initialize the network, if
 // the sidecar is enabled. If not, it returns immediately.
 func (c *Client) WaitNetworkInitialized(ctx context.Context) error {
+	err := c.syncClient.SignalEvent(ctx, runtime.NewStageNotification(c.runenv.TestGroupID, "network-initialized", "entry"))
+	if err != nil {
+		return err
+	}
+
 	if c.runenv.TestSidecar {
-		err := <-c.client.MustBarrier(ctx, "network-initialized", c.runenv.TestInstanceCount).C
+		err := <-c.syncClient.MustBarrier(ctx, "network-initialized", c.runenv.TestInstanceCount).C
 		if err != nil {
 			c.runenv.RecordMessage(InitialisationFailed)
 			return fmt.Errorf("failed to initialize network: %w", err)
 		}
 	}
 	c.runenv.RecordMessage(InitialisationSuccessful)
+
+	err = c.syncClient.SignalEvent(ctx, runtime.NewStageNotification(c.runenv.TestGroupID, "network-initialized", "exit"))
```

Comment:
Can we record `exit:err` or `exit:ok` depending on the outcome. Also, should notifications be able to carry a message? It seems like propagating the error could save us from having to look into logs.

Reply:
The idea behind notifications is for them to be `aggregated` - I don't really have a use-case for them to propagate specific messages.

`exit:err` and `exit:ok` do not belong here as far as I can tell, this is specifically the place that emits a notifications on one of the barriers we care about (and that we've had issues in the past, once networking goes down)... so I am not sure I follow the question :)